core/src/main/java/org/opensearch/sql/expression/ip/GeoipExpression.java

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.sql.expression.ip;
+
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.sql.common.grok.Converter;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionNodeVisitor;
+import org.opensearch.sql.expression.env.Environment;
+import org.opensearch.sql.planner.physical.collector.Rounding;
+
+import java.util.List;
+
+import static org.opensearch.sql.data.type.ExprCoreType.ARRAY;
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+@Getter
+@ToString
+@EqualsAndHashCode
+public class GeoipExpression implements Expression {
+    private final Expression datasource;
+    private final Expression ipAddress;
+    private final String[] properties;
+
+    public GeoipExpression(Expression datasource, Expression ipAddress, String properties) {
+        this.datasource = datasource;
+        this.ipAddress = ipAddress;
+        this.properties = properties.split(",");
```

Comment:
You probably want the parser to do this for you rather than splitting manually during logical planning.